### DISCUSSION TITLE
Micom input file

### DISCUSSION THREAD

**Discussion by Satyajitbeura** [micom-dev/micom#62] [Q&A] [answered]
Sir, I have three manually curated models. I want to make community modeling through micom but fails to import my models to micom. please help me out.

Thank you

**Comment by cdiener** [comment]
Hi, welcome to MICOM. What does your input DataFrame look like and what errors are you getting?

**Comment by Satyajitbeura** [comment]
Respected sir,
I'm new in this field. I have 3 models with me. I just made one .xls file
and now I want to lord that file to the micom for community modeling. What
are command lines I should use?

Thanking you,
Satyajit Beura

**Comment by cdiener** [comment, marked as answer]
Understood. For your Excel file make sure that you have the following column names in your table: `id`, `file`, `abundance`, `sample_id`. There should be 3 rows (excluding headers) that contain the following information:

- id: A name for the model that should not contain spaces.
- file: The path to the model file (*.xml file)
- abundance: The abundance of that organism in the sample. If you want each organism in the same abundance enter 1 here for each row.
- sample_id: It sounds you are only simulating a single sample so but the same name here for each row, for instance, "sample 1".

Now open a python session or a notebook and run the following:

```python3
import pandas as pd

table = pd.read_excel("[MY EXCEL FILE].xlsx")
```

You can have a look at the table afterwards to check everything was read correctly. You might get errors from pandas that ask you to install additional packages so just install those. After that you can use the table to run through the high-level MICOM analyses as described [in the documentation](), for instance by starting with: 

```python3
from micom.workflows import build

manifest = build(table, out_folder="models", model_db=None, threads=2)
```

If you want a more thorough introduction to running through a MICOM analysis you can also look at our [tutorials](https://micom-dev.github.io/micom/tutorials.html).

**Reply by Satyajitbeura** [reply]
Thank you sir for your help. sir, I want to design a diet (medium) instead of taking the western diet as a medium for my community model. How can I do that?

**Reply by cdiener** [reply]
You would have to identify which metabolites are present in the diet and how much of them is there. Then you need to map them to the respective ID system. For instance, if you use the AGORA database you can use https://www.vmh.life/#microbes/metabolites to map metabolite names to IDs.
